src/components/table/index.tsx

```diff
@@ -252,7 +252,11 @@ export default function EnhancedTable<T>({
                         >
                           {showCheckbox && (
                             <TableCell padding='checkbox'>
-                              <Checkbox color='primary' checked={isItemSelected} />
+                              <Checkbox
+                                color='primary'
+                                checked={isItemSelected}
+                                onClick={(e) => (!isClickable || !isClickable(row as T) ? handleClick(e, row as T) : null)}
```

Comment:
how is the click handled if the row is clickable? just curious.. the code is a bit confusing that we handleClick only if it is not-clickable.